spring-integration-file/src/main/java/org/springframework/integration/file/tail/OSDelegatingFileTailingMessageProducer.java

```diff
@@ -138,18 +134,7 @@ private void runExec() {
 		}
 	}
 
-	private TaskScheduler getRequiredTaskScheduler() {
-		if (this.scheduler == null) {
-			TaskScheduler taskScheduler = super.getTaskScheduler();
-			if (taskScheduler == null) {
-				ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
-				scheduler.initialize();
-				taskScheduler = scheduler;
-			}
-			this.scheduler = taskScheduler;
-		}
-		return this.scheduler;
```

Comment:
OK. I see where is the root of `getRequiredTaskScheduler()`. 
But a decision is the same: we always reuse `TaskScheduler` from the application context.